website/docs/src/content/cloud/storage/blob-storage.mdx

```diff
@@ -0,0 +1,287 @@
+    // Load agent memory
+    const memoryBlob = useBlob<AgentMemory>(`agents/${agentId}/memory.json`);
+    const memory = (await memoryBlob.getJSON()) ?? {
+      facts: [],
+      tasks: [],
+      lastUpdated: new Date().toISOString(),
+    };
+
+    // Process input using memory
+    // ...
+
+    // Update and save memory
+    memory.facts.push("New fact learned from input");
+    memory.tasks.push({ description: "Follow up on X", completed: false });
+    memory.lastUpdated = new Date().toISOString();
+
+    await memoryBlob.putJSON(memory);
+
+    return "Response that uses memory context";
+  },
+);
+```
+
+### Saving files
+
+You can use blob storage to save and retrieve binary files like images:
+
+```tsx
+const StoreImage = gensx.Component(
+  "StoreImage",
+  async ({ imageBuffer, filename }) => {
+    const imageBlob = useBlob(`images/${filename}`);
+
+    // Save image with metadata
+    await imageBlob.putRaw(imageBuffer, {
+      contentType: "image/png",
+      metadata: {
+        uploadedAt: new Date().toISOString(),
+        pixelSize: "800x600",
+      },
+    });
+
+    return { success: true, path: `images/${filename}` };
+  },
+);
+
+const GetImage = gensx.Component("GetImage", async ({ filename }) => {
+  const imageBlob = useBlob(`images/${filename}`);
+
+  // Check if image exists
+  const exists = await imageBlob.exists();
+  if (!exists) {
+    return { found: false };
+  }
+
+  // Get the image with metadata
+  const image = await imageBlob.getRaw();
+
+  return {
+    found: true,
+    data: image?.content,
+    contentType: image?.contentType,
+    metadata: image?.metadata,
+  };
+});
+```
+
+### Optimistic concurrency control
+
+For scenarios where multiple processes might update the same data, you can use ETags to prevent conflicts:
+
+```tsx
+const UpdateCounter = gensx.Component(
+  "UpdateCounter",
+  async ({ counterName }) => {
+    const blob = useBlob(`counters/${counterName}.json`);
+
+    // Get current value and metadata
+    const counter = (await blob.getJSON<{ value: number }>()) ?? { value: 0 };
+    const metadata = await blob.getMetadata();
+
+    // Update counter
+    counter.value += 1;
+
+    try {
+      // Save with ETag to prevent conflicts
+      await blob.putJSON(counter, {
+        etag: metadata?.etag,
+      });
+      return { success: true, value: counter.value };
+    } catch (error) {
+      if (error.name === "BlobConflictError") {
+        return {
+          success: false,
+          message: "Counter was updated by another process",
+        };
+      }
+      throw error;
+    }
+  },
+);
+```
+
+## Development vs. production
+
+GenSX blob storage works identically in both local development and cloud environments:
+
+- **Local development**: Blobs are stored in the `.gensx/blobs` directory by default
+- **Cloud deployment**: Blobs are automatically stored in cloud storage
+
+No code changes are needed when moving from development to production.
```

Comment:
Worth adding a sentence saying that if you don't specify a "kind" that we auto-infer for you based on the env.

Also - in blob this is below use cases but in database it's above; best to make it consistent